### PR TITLE
helm: table_manager configmap volume

### LINKED ISSUE #9740: No template "loki/templates/configmap.yaml" in table-manager deployment template
**Describe the bug**
Similar to #9578

https://github.com/grafana/loki/blob/main/production/helm/loki/templates/table-manager/deployment-table-manager.yaml#L110 this line says that the configmap should be called `loki.fullname`.

However the [deployed configmap](https://github.com/grafana/loki/blob/main/production/helm/loki/templates/config.yaml#L9) is `.Values.loki.externalConfigSecretName`.

If we compare the Table Manager deployment with the Read deployment (which works), this confirms that the config is wrong:
https://github.com/grafana/loki/blob/main/production/helm/loki/templates/read/deployment-read.yaml#LL134C1-L134C57

**Expected behavior**
The table manager should deploy when enabled.

**Environment:**
 - Infrastructure: n/a
 - Deployment tool: n/a

**Screenshots, Promtail config, or terminal output**
If applicable, add any output to help explain your problem.

### PR DESCRIPTION
**Pull request by tico24**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #9740

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: production/helm/loki/Chart.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ apiVersion: v2
 name: loki
 description: Helm chart for Grafana Loki in simple, scalable mode
 type: application
-appVersion: 2.8.2
-version: 5.8.0
+appVersion: 2.8.4
```
Comment (chaudum) [inline production/helm/loki/Chart.yaml:5]: This is probably a mistake. Latest Loki version is 2.8.2

**Review by chaudum — COMMENTED** [review]
Thanks for the fix! I left a comment regarding the appVersion in the Chart.yaml

File: production/helm/loki/Chart.yaml (hunk 1 above)
Comment (tico24) [reply production/helm/loki/Chart.yaml:5]: Yup. I failed to type numbers into an IDE :(

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**